src/processors/transmit.mjs

```diff
@@ -0,0 +1,97 @@
+
+let bufferLength,
+    encoderPtr,
+    samplesPtr,
+    samplesView,
+    stopped
+
+function init( {
+    activeProfileKey,
+    buffer,
+    bufferSize,
+    profiles,
+    sampleRate,
+} ) {
+
+    bufferLength = bufferSize
+    stopped = false
+    
+    const profilePtr = Module.ccall(
+        'quiet_encoder_profile_str', 'pointer', [ 'array', 'array' ], [
+            Module.intArrayFromString( JSON.stringify( profiles ) ),
+            Module.intArrayFromString( activeProfileKey )
+        ]
+    )
+    
+    encoderPtr = Module.ccall(
```

Comment:
Is it worthwhile to guard against init being called multiple times and pointers left uncleaned?

Reply:
not sure how I would guard that - this is kind of like a malloc / free cycle, it's the responsibility of the layer above to make sure it's not leaving dangling references. wdyt?